coalib/results/SourceRange.py

```diff
@@ -10,7 +10,10 @@ class SourceRange(TextRange):
     @enforce_signature
     def __init__(self,
                  start: SourcePosition,
-                 end: (SourcePosition, None)=None):
```

Comment:
This has not been well thought through.

`start` is required. So, giving `test, start_pos, end_pos` ONLY is not possible